ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -960,7 +957,14 @@ private void updateCommit(LogEntryHeader[] entriesToCommit) {
       }
       hasConfiguration |= entry.getLogEntryBodyCase() == LogEntryBodyCase.CONFIGURATIONENTRY;
       raftLog.getRaftLogMetrics().onLogEntryCommitted(entry);
+      if (entry.getLogEntryBodyCase() != LogEntryBodyCase.METADATAENTRY) {
+        lastCommitIndex = entry.getIndex();
+      }
     }
+    if (logMetadataEnabled) {
```

Comment:
It should check if `lastCommitIndex != RaftLog.INVALID_LOG_INDEX`, i.e.
```java
    if (logMetadataEnabled && lastCommitIndex != RaftLog.INVALID_LOG_INDEX) {
      logMetadata(lastCommitIndex);
    }
```

Reply:
done